src/TheThingsNetwork.cpp

```diff
@@ -234,6 +244,47 @@ int TheThingsNetwork::poll(int port, bool confirm) {
   return sendBytes(payload, 1, port, confirm);
 }
 
+void TheThingsNetwork::fillAirtimeInfo() {
+  this->info.sf = getInfo(readValue(F("radio get sf")));
+  this->info.ps = getInfo(readValue(F("radio get prlen")));
+  this->info.band = getInfo(readValue(F("radio get bw")));
+  this->info.header = getInfo(readValue(F("radio get crc")));
+  this->info.cr = getInfo(readValue(F("radio get cr")));
+  this->info.sf >= 11 ? this->info.de = 1 : this->info.de = 0;
+}
+
+int TheThingsNetwork::getInfo(String message) {
+  int i = 5;
+  int stock = 0;
+  String str;
+
+  while (i <= 8) {
+    str = "";
+    str.concat(F("4/"));
+    str.concat(i);
+    if (str == message)
+      return (i - 4);
+    i = i + 1;
+  }
+  i = -1;
+  while (message[++i]) {
+    message[i] >= '0' && message[i] <= '9' ? stock = (stock + (message[i] - 48)) * 10 : stock = stock;
+  }
+  stock == 0 && message == F("on") ? stock = 1 : stock = stock / 10;
+  return (stock);
+}
```

Comment:
OK, parse it using the pattern `SFxxBWyyy` where `xx` is either 1 or 2 digits.

What is that `"on"` doing?